libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -157,27 +187,76 @@ class FeatureFlagConfigurationSettingsConverter implements ConfigurationSettings
   Convert(
     config: object,
     options: SourceOptions
-  ): SetConfigurationSettingParam<FeatureFlagValue>[] {
-    const settings = new Array<SetConfigurationSettingParam<FeatureFlagValue>>();
+  ): SetConfigurationSettingParam[] {
+    const settings = new Array<SetConfigurationSettingParam>();
+    const featureFlags = new Array<FeatureFlagValue>();
 
-    for (const featureFlag in config) {
-      if (!this.validateFeatureName(featureFlag)) {
-        throw new ArgumentError(
-          `Feature flag ${featureFlag} contains invalid character,'%' and ':' are not allowed in feature name. Please provide valid feature name.`
-        );
+    if (this.foundMsFeatureManagement) {
+      const msFmSectionFeatureFlags = this.getMsFmSchemaSection(config);
+
+      if (msFmSectionFeatureFlags) {
+        for (const featureFlag of msFmSectionFeatureFlags) {
+          if (!featureFlag.id) {
+            throw new ArgumentError(
+              `Feature flag ${featureFlag} is in an invalid format: id is a required property`
+            );
+          }
+
+          if (featureFlags.some(existingFeatureFlag => existingFeatureFlag.id === featureFlag.id)) {
+            throw new ArgumentError(`Feature flag with id '${featureFlag.id}' already exists.`);
+          }
+
+          if (!this.validateFeatureName(featureFlag.id)) {
+            throw new ArgumentError(
+              `Feature flag ${featureFlag} contains invalid character,'%' and ':' are not allowed in feature name. Please provide valid feature id.`
+            );
+          }
+          featureFlags.push(this.getFeatureFlagDefinitionFromMsFmSchema(featureFlag));
+        }
+      }
+    }
+
+    if (this.legacySchemaFeatureManagementKeyWord) {
+      const legacySchemaFeatureFlags = this.getLegacySchema(config);
+      if (legacySchemaFeatureFlags) {
+        for (const featureFlag in legacySchemaFeatureFlags) {
+          if (!this.validateFeatureName(featureFlag)) {
+            throw new ArgumentError(
+              `Feature flag ${featureFlag} contains invalid character,'%' and ':' are not allowed in feature name. Please provide valid feature name.`
+            );
+          }
+
+          const featureManagementIndex = Constants.FeatureManagementKeyWords.indexOf(this.legacySchemaFeatureManagementKeyWord);
+
+          const featureFlagDefinition = this.getFeatureFlagDefinitionFromLegacySchema(
+            featureFlag,
+            legacySchemaFeatureFlags,
```

Comment:
Why we don't just pass the "featureFlagName" and the "featureFlagValue" which get from `legacySchemaFeatureFlags[featureFlagName]`, seems no needs to pass the whole `legacySchemaFeatureFlags`